cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90

```diff
@@ -1131,8 +1133,16 @@ subroutine ModelAdvance(gcomp, rc)
 
        write(restart_date,"(i4.4,a,i2.2,a,i2.2,a,i5.5)") yy, '-', mm, '-',dd,'-',tod
        write(restart_filename,'(4a)') trim(restart_dir), trim(restart_file), '.', trim(restart_date)
-    else
-       force_restart_now = .false.
+    endif
+
+    ! Handle end of run restart
```

Comment:
I think it is better to do the retrieval of the config variable once (in IntializeAdvertise) and use it set a logical which will enable checking if the stop alarm is ringing. Here, you are retrieving the configuration at each step through ModelAdvance. This is how it is done in CMEPS, by setting the logical flag in an _init call.

Reply:
Yep, that makes sense. Config check has been moved and code has been tested using the different config combinations.